contrib/k8s-daemonset.yaml

```diff
@@ -0,0 +1,27 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: daemonset-docker-cleanup
+spec:
+  selector:
+    matchLabels:
+      app: app-docker-cleanup
+  template:
+    metadata:
+      labels:
+        app: app-docker-cleanup
+    spec:
+      volumes:
+        - name: docker-sock
+          hostPath:
+            path: /var/run/docker.sock
+      containers:
+        - image: digirati/docker-cleanup:latest
+          name: docker-cleanup
+          env:
+            - name: CLEAN_DELAY
+              value: "86400" # 24 hours
```

Comment:
Although the possibility is slim, does Kubernetes perform a pull before rolling deployment? Could a new image be pulled and then have this script clean it up before it's used?

Reply:
:thinking: so what you're saying is:

1. Deployment of the image / manifest is triggered
2. Kubernetes pulls the new image
3. The script runs and deletes the image pulled in (2) before a new container is started using it

As far as I can tell, then yes, that's is technically a possibility. But what actually happens is that one of pods running the image will be destroyed, and as it comes back up it pulls the new image. There is a fraction of a moment between the image being pulled, and the container starting, so the likelihood of that happening in reality seems so narrow it's almost not worth considering.

If that did happen, I believe Kubernetes would recover by simply pulling the image again. It would only become an issue if someone set `CLEAN_DELAY` to something ridiculously low, e.g. 1 second.

Reply:
Ah cool - that recovery makes sense 👍  That's was indeed my thought process! Sounds good